lib/smart_proxy_container_gateway/container_gateway_api.rb

```diff
@@ -100,18 +104,24 @@ class Api < ::Sinatra::Base
         response.headers['Docker-Distribution-API-Version'] = 'registry/2.0'
 
         unless auth_header.present? && auth_header.basic_auth?
-          one_year = (60 * 60 * 24 * 365)
-          return { token: AuthorizationHeader::UNAUTHORIZED_TOKEN, issued_at: Time.now.iso8601,
-                   expires_at: (Time.now + one_year).iso8601 }.to_json
+          return { token: AuthorizationHeader::UNAUTHORIZED_TOKEN, issued_at: Time.now.rfc3339,
+                   expires_in: 1.year.seconds.to_i }.to_json
         end
 
         token_response = ForemanApi.new.fetch_token(auth_header.raw_header, request.params)
         if token_response.code.to_i != 200
           halt token_response.code.to_i, token_response.body
         else
+          # This returned token should follow OAuth2 spec. We need some minor conversion
+          # to store the token with the expires_at time (using rfc3339).
           token_response_body = JSON.parse(token_response.body)
-          ContainerGateway.insert_token(request.params['account'], token_response_body['token'],
-                                        token_response_body['expires_at'])
+          issued_at = token_response_body["issued_at"].to_time
+          expires_at = issued_at + token_response_body["expires_in"].seconds
```

Comment:
The [spec](https://distribution.github.io/distribution/spec/auth/token/#requesting-a-token) says both are optional. At least one can be easily dealt with since it should assume 60 when missing so you can use `.fetch()`.